### PR TITLE
Save User and Redirect on Successful Login

### PR DESCRIPTION
**Pull request by MKiperszmid**

### CODE REVIEW

File: app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginScreen.kt [hunk 1]
```diff
@@ -16,16 +17,26 @@ import com.mk.tasky.R
...
+    onLogin: () -> Unit,
     viewModel: LoginViewModel = hiltViewModel()
 ) {
     val state = viewModel.state
 
+    LaunchedEffect(key1 = true) {
+        viewModel.uiEvent.collect {
+            if (it is UIEvent.Navigate) {
+                onLogin()
+            }
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginScreen.kt]: Around the end of yesterdays meeting (recording is online), I talked about how you could model such events as state. I personally stopped using such event channels/flows after an interesting talk I saw at Droidcon. I did it like this here for a long time as well (and the UIEvent class also feels very familiar :D ), but now I feel like that doesn't fit that well into a fully state-driven environment like Compose.

But that's just my view, if you like using such events, there's nothing wrong about it as long as you correctly collect them. Just wanted throw that in :)

File: app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginViewModel.kt [hunk 2]
```diff
@@ -62,7 +70,10 @@ class LoginViewModel @Inject constructor(
         viewModelScope.launch {
             try {
                 repository.login(email, password).onSuccess {
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginViewModel.kt]: Was this the try/catch block I commented on previously or another one? It's quite unnecessary

File: app/src/main/java/com/mk/tasky/core/di/CoreModule.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    @Provides
+    @Singleton
+    fun providePreferences(sharedPreferences: SharedPreferences): Preferences {
+        return DefaultPreferences(sharedPreferences)
+    }
+
+    @Provides
+    @Singleton
+    fun provideSharedPreferences(app: Application): SharedPreferences {
+        return app.getSharedPreferences("shared_prefs", Context.MODE_PRIVATE)
```
Comment (philipplackner) [inline app/src/main/java/com/mk/tasky/core/di/CoreModule.kt]: It's strongly recommended to use `EncryptedSharedPreferences` from the Jetpack crypto library, if you save sensitive things like a JWT token. Anybody who gets access to that can perform API calls in your name.

**Review by philipplackner — APPROVED** [review]

File: app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginViewModel.kt (hunk 2 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginViewModel.kt]: Oh nono, I had 2. One for Login and one for Registration
We removed the registration one, but totally forgot about Login

Nice catch!

File: app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginScreen.kt (hunk 1 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/authentication/login/presentation/LoginScreen.kt]: Will check the Q&A's recording to learn about it. Thanks!

File: app/src/main/java/com/mk/tasky/core/di/CoreModule.kt (hunk 3 above)
Comment (MKiperszmid) [reply app/src/main/java/com/mk/tasky/core/di/CoreModule.kt]: Will investigate about it. Thanks!!

**Pull request merged**